source/examples/gsfa/GSFA.rst

```diff
@@ -0,0 +1,412 @@
+.. _gsfa:
+
+======================
+Graph-Based SFA (GSFA)
+======================
+.. codesnippet::
+
+-  Extension of slow feature analysis
+   `(SFA) <https://www.ini.rub.de/research/blog/a_brief_introduction_to_slow_feature_analysis/>`__
+
+-  Supervised dimensionality reduction method
+
+-  Trained with a graph, in which the vertices are the samples and the
+   edges represent similarities of the corresponding labels.
+
+Graph-based slow feature analysis (GSFA) is a supervised extension of
+SFA algorithm that relies on particular graphs structure to extract
+features that preserve label similarities. More precisely, the algorithm
+utilize training graphs, in which the verticies are the samples and the
+edges represent similarities of the corresponding labels. Later, we use
+aquired low-dimensional representaion of the original data to train
+typical supervised learning algortihms.
+
+In this example we will briefly exlpain the idea behind GSFA and
+specify the optimization task it solves. Moreover, we will show the
+effeciency of the GSFA compared to SVM on a toy dataset and introduce an
```

Comment:
I know, grammerly suggested to add "the" before GSFA here (or was it somewhere else?). In case of GSFA, however, grammerly mistakes it to be a thing rather than a name. Since it actually is a name, there should be no "the", unless you write "the GSFA algorithm" or something. On the other hand I feel like there should be a "the" before "SVM" because the "M" stands for "machine" which is a thing. Certainly, in the line below it should either be "the classification task" or "classification tasks" (in plural but without "the").